Expensify/App
# Add new sign in flow

# NOTE
This is not hooked up to any APIs yet, it just walks through the form steps.

Things left to do:
- [x] Call `GetAccountStatus` to see if the user had an account already and if they have access to e.cash
- [x] Call `SetGithubUsername` to save the github username
- [x] Call `CreateLogin` if account already existed
- [x] Call `CreateAccount` if account didn't exist
- [x] See if the 2FA field can be hidden if it's not required (maybe add this to `GetAccountStatus`?)

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/148719
Fixes https://github.com/Expensify/Expensify/issues/148714
Fixes https://github.com/Expensify/Expensify/issues/148716

### Tests
As you go through the testing, verify that form validation works (you can't submit an empty form)

1. Fill out the login form, verify you are taken to the github username form
2. Fill out the github username form, verify you are taken to the password form
3. Fill out the password form, verify you are taken to the resend validation link form
4. Clear out local storage according to [this SO](https://stackoverflow.com/c/expensify/questions/6923/6924#6924)
5. Refresh the page and verify you are back at the first login form

### Screenshots
|![image](https://user-images.githubusercontent.com/1228807/102694076-f6d92000-41db-11eb-845a-7e3a972856cb.png)|![image](https://user-images.githubusercontent.com/1228807/102694078-f93b7a00-41db-11eb-98fa-3904803a441e.png)
|![image](https://user-images.githubusercontent.com/1228807/102694081-fb9dd400-41db-11eb-8209-ac8709a8981d.png)|![image](https://user-images.githubusercontent.com/1228807/102694066-e6c14080-41db-11eb-9e85-8a42a296f4c5.png)



cc @chiragsalian @stitesExpensify @shawnborton
